packages/loopring_v3/contracts/impl/libtransactions/AccountUpdateTransaction.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.6.10;
+pragma experimental ABIEncoderV2;
+
+import "../../iface/ExchangeData.sol";
+import "../../thirdparty/BytesUtil.sol";
+import "../../lib/EIP712.sol";
+import "../../lib/FloatUtil.sol";
+import "../../lib/SignatureUtil.sol";
+
+
+/// @title AccountUpdateTransaction
+/// @author Brecht Devos - <brecht@loopring.org>
+library AccountUpdateTransaction
+{
+    using BytesUtil            for bytes;
+    using FloatUtil            for uint;
+    using SignatureUtil        for bytes32;
+
+    bytes32 constant public ACCOUNTUPDATE_TYPEHASH = keccak256(
+        "AccountUpdate(address owner,uint24 accountID,uint32 nonce,uint256 publicKey,uint256 walletHash,uint16 feeTokenID,uint256 fee)"
+    );
+
+    /*event AccountUpdated(
+        uint24 owner,
+        uint   publicKey
+    );*/
+
+    function process(
+        ExchangeData.State        storage S,
+        ExchangeData.BlockContext memory  ctx,
+        bytes                     memory  data,
+        bytes                     memory  auxiliaryData
+        )
+        internal
+        returns (uint /*feeETH*/)
+    {
+        uint offset = 1;
```

Comment:
Do we need to require the minimum data length for each transaction?

Reply:
No need, the block data posted onchain needs to match perfectly with the public data the circuit expects, otherwise the hash is different.